packages/@aws-cdk/aws-ec2/lib/network-acl.ts

```diff
@@ -0,0 +1,411 @@
+      networkAclId: this.networkAcl.networkAclId,
+      ruleNumber: props.ruleNumber,
+      protocol: props.protocol,
+      ruleAction: props.ruleAction,
+      egress: props.egress,
+      cidrBlock: props.cidrBlock,
+      icmp: props.icmp,
+      portRange: props.portRange
+    });
+
+  }
+
+  public get uniqueId() {
+    return this.node.uniqueId;
+  }
+}
+
+export interface ISubnetNetworkAclAssociation extends IResource {
+  /**
+   * ID for the current SubnetNetworkAclAssociation
+   * @attribute
+   */
+  readonly subnetNetworkAclAssociationAssociationId: string;
+}
+
+/**
+ * A SubnetNetworkAclAssociationBase that is not created in this template
+ */
+abstract class SubnetNetworkAclAssociationBase extends Resource implements ISubnetNetworkAclAssociation {
+public abstract readonly subnetNetworkAclAssociationAssociationId: string;
+constructor(scope: Construct, id: string, props?: ResourceProps) {
+    super(scope, id, props);
+
+  }
+
+  public get uniqueId() {
+    return this.node.uniqueId;
+  }
+}
+
+export interface SubnetNetworkAclAssociationProps {
+  /**
+   * The name of the SubnetNetworkAclAssociation.
+   *
+   * It is not recommended to use an explicit name.
+   *
+   * @default If you don't specify a SubnetNetworkAclAssociationName, AWS CloudFormation generates a
+   * unique physical ID and uses that ID for the group name.
+   */
+  readonly subnetNetworkAclAssociationName?: string;
+
+  /**
+   * ID for the current Network ACL
+   * @attribute
+   */
+  readonly networkAclId: string;
+  /**
+   * ID of the Subnet
+   * @attribute
+   */
+  readonly subnetId: string;
+}
+
+export class SubnetNetworkAclAssociation extends SubnetNetworkAclAssociationBase {
+  /**
+   * Import an existing NetworkAcl into this app.
+   */
+  public static fromSubnetNetworkAclAssociationId(scope: Construct, id: string,
+                                                  subnetNetworkAclAssociationId: string): ISubnetNetworkAclAssociation {
+  class Import extends SubnetNetworkAclAssociationBase {
+    public subnetNetworkAclAssociationAssociationId = subnetNetworkAclAssociationId;
+
+    }
+
+  return new Import(scope, id);
+  }
+  /**
+   * ID for the current SubnetNetworkAclAssociation
+   * @attribute
+   */
+
+  public readonly subnetNetworkAclAssociationAssociationId: string;
+  /**
+   * ID for the current Network ACL
+   * @attribute
+   */
+  public readonly networkAclId: string;
+  /**
+   * ID of the Subnet
+   * @attribute
+   */
+  public readonly subnetId: string;
+
+  private association: CfnSubnetNetworkAclAssociation;
+  constructor(scope: Construct, id: string, props: SubnetNetworkAclAssociationProps) {
+    super(scope, id, {
+      physicalName: props.subnetNetworkAclAssociationName
+    });
+
+    this.association = new CfnSubnetNetworkAclAssociation(this, 'Resource', {
+       networkAclId: props.networkAclId,
+       subnetId: props.subnetId,
+    });
+
+    this.networkAclId = this.association.networkAclId;
```

Comment:
Don't think this is correct. Shouldn't it be `this.association.attrAclId` ? And also for all other cases.

Reply:
this.association.attrAclId does not exist ,
The only one that starts with attr is this.association.attrAssociationId;